py/server/tests/test_pandas.py

```diff
@@ -296,19 +298,46 @@ def test_numpy_array(self):
 
     def test_pandas_category_type(self):
         df = pd.DataFrame({
-                              'zipcode': {17384: 98125, 2680: 98107, 722: 98005, 18754: 98109, 14554: 98155},
-                              'bathrooms': {17384: 1.5, 2680: 0.75, 722: 3.25, 18754: 1.0, 14554: 2.5},
-                              'sqft_lot': {17384: 1650, 2680: 3700, 722: 51836, 18754: 2640, 14554: 9603},
-                              'bedrooms': {17384: 2, 2680: 2, 722: 4, 18754: 2, 14554: 4},
-                              'sqft_living': {17384: 1430, 2680: 1440, 722: 4670, 18754: 1130, 14554: 3180},
-                              'floors': {17384: 3.0, 2680: 1.0, 722: 2.0, 18754: 1.0, 14554: 2.0}
-                          })
+            'zipcode': {17384: 98125, 2680: 98107, 722: 98005, 18754: 98109, 14554: 98155},
+            'bathrooms': {17384: 1.5, 2680: 0.75, 722: 3.25, 18754: 1.0, 14554: 2.5},
+            'sqft_lot': {17384: 1650, 2680: 3700, 722: 51836, 18754: 2640, 14554: 9603},
+            'bedrooms': {17384: 2, 2680: 2, 722: 4, 18754: 2, 14554: 4},
+            'sqft_living': {17384: 1430, 2680: 1440, 722: 4670, 18754: 1130, 14554: 3180},
+            'floors': {17384: 3.0, 2680: 1.0, 722: 2.0, 18754: 1.0, 14554: 2.0}
+        })
         df['zipcode'] = df.zipcode.astype('category')
         df['bathrooms'] = df.bathrooms.astype('category')
         t = to_table(df)
         self.assertEqual(t.columns[0].data_type, dtypes.int64)
         self.assertEqual(t.columns[1].data_type, dtypes.double)
 
+    def test_conv_null(self):
+        input_cols = [
+            bool_col(name="Boolean", data=(True, NULL_BOOLEAN)),
+            byte_col(name="Byte", data=(1, NULL_BYTE)),
+            char_col(name="Char", data=(1, NULL_CHAR)),
+            short_col(name="Short", data=[1, NULL_SHORT]),
+            int_col(name="Int_", data=[1, NULL_INT]),
+            long_col(name="Long_", data=[1, NULL_LONG]),
+            float_col(name="Float_", data=[np.nan, NULL_FLOAT]),
+            double_col(name="Double_", data=[np.nan, NULL_DOUBLE]),
+            datetime_col(name="Datetime", data=[epoch_nanos_to_instant(1), None]),
+        ]
+        t = new_table(cols=input_cols)
+        df = to_pandas(t, conv_null=True)
+        dh_table = to_table(df)
+        # Round-trip is not possible when floating point data has NaN because NaN becomes pd.NA during conversion
+        # self.assertFalse(table_equals(t.head(1), dh_table.head(1)))
+        # self.assert_table_equals(t.tail(1), dh_table.tail(1))
```

Comment:
See comments about adding an argument that allows users to specify NaN handling.